pages/api/neon-prisma-global.ts

```diff
@@ -2,24 +2,30 @@ import { NextRequest as Request, NextResponse as Response } from "next/server";
 import { Pool } from '@neondatabase/serverless'
 import { PrismaNeon } from '@prisma/adapter-neon'
 import { PrismaClient } from '@/prisma-neon/prisma-client'
+import { RequestContext } from "@vercel/edge";
 
 export const config = {
   runtime: "edge",
 };
-const pool = new Pool({ connectionString: process.env.NEON_DATABASE_URL, idleTimeoutMillis: 1 })
-const adapter = new PrismaNeon(pool)
-const prisma = new PrismaClient({ adapter })
+
 const start = Date.now()
 
-export default async function api(req: Request) {
+export default async function api(req: Request, ctx: RequestContext) {
   const count = toNumber(new URL(req.url).searchParams.get("count"));
   const time = Date.now();
 
+  const pool = new Pool({ connectionString: process.env.NEON_DATABASE_URL })
+  const adapter = new PrismaNeon(pool)
+  const prisma = new PrismaClient({ adapter })
+
   let data = null;
   for (let i = 0; i < count; i++) {
 
     data = await prisma.employees.findMany({ take: 10 });
   }
+
+  ctx.waitUntil(pool.end());
```

Comment:
Could you use `@vercel/functions` instead? It's isomorphic 

```
import { waitUntil } from '@vercel/functions';
 
export function GET() {
  const country = request.headers.get('x-vercel-ip-country');
  // Return a response immediately, but then afterwards call the fetch
  waitUntil(fetch(`https://api.vercel.app/countries/?incr=${country}`));
  return new Response(`You're visiting from beautiful ${country}`);
}
```

https://vercel.com/docs/functions/functions-api-reference#waituntil

Reply:
Ah, nice one TIL. Thanks @leerob

Reply:
It's brand new!